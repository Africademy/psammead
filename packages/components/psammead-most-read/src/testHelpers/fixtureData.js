export const itemsLTR = [
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];

export const itemsRTL = [
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];

export const links = [
  {
    language: 'news',
    title: 'John Lewis staff bonus cut again as profits fall',
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
  {
    language: 'arabic',
    title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
    href: 'https://www.bbc.com/vietnamese/institutional-49283563',
  },
];
