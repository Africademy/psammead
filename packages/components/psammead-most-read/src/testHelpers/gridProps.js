const mostReadListGridProps = maxTwoColumns => ({
  enableGelGutters: true,
  enableGelMargins: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 8,
    group5: maxTwoColumns ? 8 : 20,
  },
});

const mostReadItemGridProps = {
  item: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 3,
    group4: 4,
    group5: 4,
  },
};

export { mostReadListGridProps, mostReadItemGridProps };
