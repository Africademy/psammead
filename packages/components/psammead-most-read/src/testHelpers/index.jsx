import React from 'react';
import { shape, string, oneOf, number } from 'prop-types';
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import { scriptPropType } from '@bbc/gel-foundations/prop-types';
import Timestamp from '@bbc/psammead-timestamp';
import {
  MostReadTitle,
  MostReadList,
  MostReadItemWrapper,
  MostReadLink,
  MostReadRank,
} from '..';

const lastUpdated = (script, service) => (
  // This will return the provided english translations
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

export const getItem = (service, withTimestamp = false) => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  const timestamp = withTimestamp ? lastUpdated(latin, service) : null;
  return {
    id: Math.floor(Math.random() * 100000) + 1,
    title: text,
    href: `${baseUrl}${articlePath}`,
    timestamp,
  };
};

export const getItemWrapperArray = ({
  numberOfItems,
  service,
  script,
  dir,
}) => {
  const itemWrapperArray = [];

  for (let i = 1; i <= numberOfItems; i += 1) {
    itemWrapperArray.push(
      <MostReadItemWrapper dir={dir} key={i}>
        <MostReadRank
          service={service}
          script={script}
          listIndex={i}
          numberOfItems={numberOfItems}
          dir={dir}
        />
        <MostReadLink
          dir={dir}
          href={getItem(service).href}
          service={service}
          script={script}
          title={getItem(service).title}
        />
      </MostReadItemWrapper>,
    );
  }
  return itemWrapperArray;
};

export const renderMostRead = ({
  service,
  script,
  dir,
  header,
  numberOfItems,
}) => (
  <>
    <MostReadTitle
      header={header}
      service={service}
      script={script}
      dir={dir}
    />
    <MostReadList numberOfItems={numberOfItems} dir={dir}>
      {getItemWrapperArray({
        numberOfItems,
        service,
        script,
        dir,
      }).map(item => item)}
    </MostReadList>
  </>
);

renderMostRead.propTypes = {
  service: string.isRequired,
  script: shape(scriptPropType).isRequired,
  dir: oneOf(['rtl', 'ltr']),
  header: string.isRequired,
  numberOfItems: number.isRequired,
};

renderMostRead.defaultProps = {
  dir: 'ltr',
};
