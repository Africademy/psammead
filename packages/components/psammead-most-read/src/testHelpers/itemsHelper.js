import React from 'react';
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { shape, string, number } from 'prop-types';
import Timestamp from '@bbc/psammead-timestamp';

const lastUpdated = (script, service) => (
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

export const getItem = service => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    id: Math.random(),
    title: text,
    href: `${baseUrl}${articlePath}`,
  };
};

export const getItemWithTimestamp = (script, service) => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    id: Math.random(),
    title: text,
    href: `${baseUrl}${articlePath}`,
    timestamp: lastUpdated(script, service),
  };
};

export const getItems = (service = 'news', number) =>
  Array(number).fill(getItem(service));

export const itemPropTypes = shape({
  id: number,
  title: string,
  href: string,
  timestamp: number,
});
