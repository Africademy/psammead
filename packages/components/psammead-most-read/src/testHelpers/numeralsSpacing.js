const notBengali = {
  minWidth: '3rem',

}