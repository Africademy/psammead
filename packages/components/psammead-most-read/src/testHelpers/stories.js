import { withServicesKnob } from '@bbc/psammead-storybook-helpers';

export const arabicServiceDecorator = withServicesKnob({
  defaultService: 'arabic',
});

export const bengaliServiceDecorator = withServicesKnob({
  defaultService: 'bengali',
});

export const burmeseServiceDecorator = withServicesKnob({
  defaultService: 'burmese',
});

export const newsServiceDecorator = withServicesKnob({
  defaultService: 'news',
});
