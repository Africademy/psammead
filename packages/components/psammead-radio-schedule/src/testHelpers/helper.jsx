/* eslint-disable react/prop-types */
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { arabic, latin } from '@bbc/gel-foundations/scripts';
import React from 'react';
import ProgramCard from '../ProgramCard';
import RadioSchedule from '../index';

export const sentenceCase = text =>
  text
    .toLowerCase()
    .charAt(0)
    .toUpperCase() + text.substring(1);

export const stateTypes = ['onDemand', 'live', 'next'];

const getSchedule = (service, withLongSummary) => {
  const { text, articlePath, longText } = TEXT_VARIANTS[service];

  return stateTypes.map((state, index) => ({
    id: index,
    state,
    stateLabel: sentenceCase(state),
    startTime: 1566914061212,
    link: articlePath,
    brandTitle: text,
    episodeTitle: '29/01/1990',
    summary:
      withLongSummary && state === 'live'
        ? `${longText} ${longText}`
        : longText,
    duration: 'PT45M',
    durationLabel: 'Duration',
  }));
};

export const renderProgramCard = ({
  state,
  stateLabel,
  service = 'news',
  episodeTitle = '29/01/1990',
  duration = 'PT30M',
  durationLabel = 'Duration',
  startTime = 1566914061212,
}) => {
  const { text, articlePath, longText, dir, locale, timezone } = TEXT_VARIANTS[
    service
  ];
  const offscreenLiveText = state === 'live' ? 'Live' : null;
  return (
    <ProgramCard
      service={service}
      script={dir === 'rtl' ? arabic : latin}
      dir={dir}
      brandTitle={text}
      summary={longText}
      episodeTitle={episodeTitle}
      duration={duration}
      startTime={startTime}
      state={state}
      link={articlePath}
      durationLabel={durationLabel}
      stateLabel={stateLabel}
      timezone={timezone}
      locale={locale}
      offscreenLiveText={offscreenLiveText}
    />
  );
};

export const renderRadioSchedule = ({
  service = 'news',
  locale = 'en-gb',
  timezone = 'Europe/London',
  script = latin,
  dir = 'ltr',
  withLongSummary = false,
}) => (
  <RadioSchedule
    schedules={getSchedule(service, withLongSummary)}
    locale={locale}
    timezone={timezone}
    script={script}
    service={service}
    dir={dir}
  />
);
