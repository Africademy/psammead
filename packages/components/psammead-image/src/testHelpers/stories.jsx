import React from 'react';
import { storiesOf } from '@storybook/react';
import Readme from '../../README.md';
import { custom, landscape, portrait, square } from './fixtureData';

function getProps(image, includeHeight) {
  const props = {
    alt: image.alt,
    src: image.src,
    srcset: image.srcset,
    width: image.width,
  };
  props.height = includeHeight ? image.height : null;

  return props;
}

const stories = (
  Component,
  title,
  includeHeight = false,
  additionalProps = {},
  styleDecorator = storyFn => storyFn(),
) =>
  storiesOf(title, module)
      .addDecorator(styleDecorator)
    .add('landscape image', () => (
      <Component {...getProps(landscape, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('portrait image', () => (
      <Component {...getProps(portrait, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('square image', () => (
      <Component {...getProps(square, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('custom ratio image', () => (
      <Component {...getProps(custom, includeHeight)} {...additionalProps} />
    ), { notes: Readme })
    .add('image with srcset', () => (
      <Component
        {...getProps(landscape, includeHeight)}
        srcset={landscape.srcset}
        {...additionalProps}
      />
    ), { notes: Readme });

export default stories;
