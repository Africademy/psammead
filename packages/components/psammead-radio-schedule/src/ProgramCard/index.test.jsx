import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { renderProgramCard } from '../testHelpers/helper';

const stateTypes = ['live', 'next', 'onDemand'];

describe('ProgramCard', () => {
  stateTypes.forEach(type => {
    shouldMatchSnapshot(
      `should render correctly for ${type}`,
      renderProgramCard({ state: { type, translation: type } }),
    );
  });

  shouldMatchSnapshot(
    `should render correctly in RTL`,
    renderProgramCard({
      state: { type: stateTypes[0], translation: 'مباشر' },
      duration: { durationValue: '30:00', durationText: 'المدة الزمنية' },
      service: 'arabic',
    }),
  );
});
