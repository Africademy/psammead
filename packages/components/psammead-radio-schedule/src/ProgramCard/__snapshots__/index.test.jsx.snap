// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramCard should render correctly for live 1`] = `
.c9 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c2 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:visited {
  color: #6E6E73;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  position: relative;
}

.c1 {
  padding: 0 0.5rem;
}

.c3 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  margin: 0;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  text-transform: uppercase;
  color: #B80000;
}

.c5 {
  color: #3F3F42;
  padding: 0.5rem 0;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c7 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #B80000;
  color: #FFFFFF;
}

.c8 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c10 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/news/articles/cn7k01xp8kxo"
    >
      <h3
        class="c3"
      >
        <span
          class="c4"
        >
          live 
        </span>
        Could a computer ever create better art than a human?
      </h3>
    </a>
    <div
      class="c5"
    >
      29/01/1990
    </div>
    <p
      class="c6"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c7"
  >
    <span
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c9"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <span
      class="c10"
      dir="ltr"
    >
      30:00
    </span>
  </div>
</div>
`;

exports[`ProgramCard should render correctly for next 1`] = `
.c9 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c2 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:visited {
  color: #6E6E73;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  position: relative;
}

.c1 {
  padding: 0 0.5rem;
}

.c3 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  margin: 0;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  text-transform: uppercase;
  color: #11708C;
}

.c5 {
  color: #6E6E73;
  padding: 0.5rem 0;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c7 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #FFFFFF;
  color: #11708C;
}

.c8 > svg {
  color: #11708C;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c10 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/news/articles/cn7k01xp8kxo"
    >
      <h3
        class="c3"
      >
        <span
          class="c4"
        >
          next 
        </span>
        Could a computer ever create better art than a human?
      </h3>
    </a>
    <div
      class="c5"
    >
      29/01/1990
    </div>
    <p
      class="c6"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c7"
  >
    <span
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c9"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <span
      class="c10"
      dir="ltr"
    >
      30:00
    </span>
  </div>
</div>
`;

exports[`ProgramCard should render correctly for onDemand 1`] = `
.c8 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c2 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:visited {
  color: #6E6E73;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  position: relative;
}

.c1 {
  padding: 0 0.5rem;
}

.c3 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  margin: 0;
}

.c4 {
  color: #3F3F42;
  padding: 0.5rem 0;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c5 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #222222;
  color: #FFFFFF;
}

.c7 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c9 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/news/articles/cn7k01xp8kxo"
    >
      <h3
        class="c3"
      >
        Could a computer ever create better art than a human?
      </h3>
    </a>
    <div
      class="c4"
    >
      29/01/1990
    </div>
    <p
      class="c5"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c6"
  >
    <span
      class="c7"
    >
      <svg
        aria-hidden="true"
        class="c8"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <span
      class="c9"
      dir="ltr"
    >
      30:00
    </span>
  </div>
</div>
`;

exports[`ProgramCard should render correctly in RTL 1`] = `
.c9 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c2 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:visited {
  color: #6E6E73;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  position: relative;
}

.c1 {
  padding: 0 0.5rem;
}

.c3 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.625rem;
  color: #222222;
  margin: 0;
}

.c4 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.625rem;
  text-transform: uppercase;
  color: #B80000;
}

.c5 {
  color: #3F3F42;
  padding: 0.5rem 0;
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.625rem;
}

.c6 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1.0625rem;
  line-height: 1.5rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c7 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #B80000;
  color: #FFFFFF;
}

.c8 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c10 {
  padding-right: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/arabic/articles/c1er5mjnznzo"
    >
      <h3
        class="c3"
      >
        <span
          class="c4"
        >
          مباشر 
        </span>
        لماذا يخجل البعض من اسم قريته في مصر؟
      </h3>
    </a>
    <div
      class="c5"
    >
      29/01/1990
    </div>
    <p
      class="c6"
    >
      هناك وقائع عدة تتسم بالسخرية والجدل والتنمر، ضد أهل القرية الذين أصابهم الغضب والسخط مما دفعهم إلى تقديم طلب لتغيير اسم قريتهم.
    </p>
  </div>
  <div
    class="c7"
  >
    <span
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c9"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <span
      class="c10"
      dir="rtl"
    >
      30:00
    </span>
  </div>
</div>
`;
